dev/src/write-batch.ts

```diff
@@ -479,13 +479,26 @@ export class WriteBatch {
       if (!document.isEmpty || !documentMask.isEmpty) {
         write = document.toProto();
         write!.updateMask = documentMask.toProto();
-      }
 
-      return {
-        write,
-        transform: transform.toProto(this._serializer),
-        precondition: precondition.toProto(),
-      };
+        const transformProto = transform.toProto(this._serializer);
```

Comment:
Can we update `transform.toProto` to return `transformProto.transform.fieldTransforms`? Ideally we could just write:

```
write = document.toProto();
write!.updateMask = documentMask.toProto();	        
write!. updateTransforms = transform.toProto();
```

```

Reply:
It's currently being used that way in `set` and `create`, so I didn't change it. I could also write it as a separate method like `toFieldTransformProto`, but that seems like extra code. Thoughts?

Reply:
You should replace all usages of `transform` with `write. updateTransforms`. If you don't, you will get super confused later on. Trust me :)

Reply:
done. grasping for the light